invenio_rdm_records/resources/resources.py

```diff
@@ -89,16 +93,30 @@ def p(route):
     @request_extra_args
     @request_read_args
     @request_view_args
-    @response_handler()
+    @custom_response_handler()
     def read(self):
         """Read an item."""
-        item = self.service.read(
-            g.identity,
-            resource_requestctx.view_args["pid_value"],
-            expand=resource_requestctx.args.get("expand", False),
-            # allows to access deleted record if permissions match
-            include_deleted=resource_requestctx.args.get("include_deleted", False),
-        )
+        try:
+            item = self.service.read(
+                g.identity,
+                resource_requestctx.view_args["pid_value"],
+                expand=resource_requestctx.args.get("expand", False),
+                # allows to access deleted record if permissions match
+                include_deleted=resource_requestctx.args.get("include_deleted", False),
+            )
+        except NoResultFound:
+            # If the parent pid is being used we can get the id of the latest record and redirect
+            latest_version = self.service.read_latest(
+                g.identity,
+                resource_requestctx.view_args["pid_value"],
+                expand=resource_requestctx.args.get("expand", False),
+            )
+            return redirect(
+                url_for(
+                    "records.read",
```

Comment:
```suggestion
                    ".",
```

This might work, or `.read` as well. This is to safeguard in case we rename/move the blueprint.